Source/WebCore/rendering/RenderTable.cpp

```diff
@@ -1508,9 +1508,10 @@ RenderTableCell* RenderTable::cellAfter(const RenderTableCell* cell) const
 
 LayoutUnit RenderTable::baselinePosition(FontBaseline baselineType, bool firstLine, LineDirectionMode direction, LinePositionMode linePositionMode) const
 {
-    return valueOrCompute(firstLineBaseline(), [&] {
-        return RenderBox::baselinePosition(baselineType, firstLine, direction, linePositionMode);
-    });
+    auto baselinePos = firstLineBaseline();
+    if (baselinePos)
+        return (direction == HorizontalLine ? marginTop() : marginRight()) + firstLineBaseline().value();
```

Comment:
I get that you need this fix for some tests, but it's actually fixing the reference, not the flexbox tests, so it seems strange to do it here.
I would prefer to do this change in its own PR, and have a specific test not involving flexbox, like comparing
```html
<div style="display: inline-block; background: lime;">line 1</div>
<div style="display: inline-table; margin-top: 20px; background: aqua;">line 1<br/>line 2</div>
```
with this reference:
```html
<div style="display: inline-block; margin-top: 20px; background: lime; vertical-align: top">line 1</div>
<div style="display: inline-block; margin-top: 20px; background: aqua;">line 1<br/>line 2</div>
```

Reply:
That makes sense. I refactored it out and created its own PR 

https://github.com/WebKit/WebKit/pull/5153